src/test/java/com/vmware/vipclient/i18n/messages/service/CacheServiceTest.java

```diff
@@ -225,4 +229,26 @@ public void testExpireUsingCacheExpiredTimeConfig() throws InterruptedException
         responseTime2 = cacheItem.getTimestamp();
         assertTrue(responseTime2 > responseTime); 
     }
+
+    @Test
+    public void testGetSupportedLocalesOfflineBundles() {
+        //Enable offline mode
+        String offlineResourcesBaseUrlOrig = cfg.getOfflineResourcesBaseUrl();
+        cfg.setOfflineResourcesBaseUrl("offlineBundles/");
+        List<DataSourceEnum> msgOriginsQueueOrig = cfg.getMsgOriginsQueue();
+        cfg.setMsgOriginsQueue(new LinkedList<>(Arrays.asList(DataSourceEnum.Bundle)));
+
+        cfg.createTranslationCache(MessageCache.class);
+        cfg.initializeMessageCache();
+
+        CacheService cs = new CacheService(new MessagesDTO());
+        List<Locale> supportedLocales = cs.getSupportedLocalesFromCache();
+        assertTrue(supportedLocales.contains(Locale.forLanguageTag("fil")));
+        assertEquals("Filipino", supportedLocales.get(
+                supportedLocales.indexOf(Locale.forLanguageTag("fil"))).getDisplayName());
+
+        // Disable offline mode off for next tests.
+        cfg.setOfflineResourcesBaseUrl(offlineResourcesBaseUrlOrig);
```

Comment:
To make sure these 2 lines can be performed, above code should be in catch block.